src/renku_crc/server_options.py

```diff
@@ -0,0 +1,167 @@
+    options: List[int] = Field(min_items=1)
+
+    class Config:
+        extra = Extra.ignore
+
+    @validator("options", pre=False, each_item=True)
+    def greater_than_or_equal_to_zero(cls, v):
+        if v < 0:
+            raise ValueError(f"The provided value should be greater than or equal to zero, instead it was {v}.")
+        return v
+
+
+class _ServerOptionsBytes(BaseModel):
+    options: List[ByteSize] = Field(min_items=1)
+
+    class Config:
+        extra = Extra.ignore
+
+    @validator("options", pre=True)
+    def convert_units(cls, vals):
+        for ival, val in enumerate(vals):
+            if isinstance(val, str) and val.strip().endswith("i"):
+                vals[ival] = val.strip() + "b"
+        return vals
+
+    @validator("options", pre=False, each_item=True)
+    def greater_than_zero(cls, val):
+        return _check_greater_than_zero(cls, val)
+
+
+class ServerOptions(BaseModel):
+    """Used to parse the server options passed to the notebook service in the Helm values."""
+
+    cpu_request: _ServerOptionsCpu
+    mem_request: _ServerOptionsBytes
+    disk_request: _ServerOptionsBytes
+    gpu_request: _ServerOptionsGpu = Field(default_factory=lambda: _ServerOptionsGpu(options=[0]))
+
+    class Config:
+        """Configuration."""
+
+        extra = Extra.ignore
+
+    def find_largest_attribute(self) -> str:
+        """Find the attribute with the largest number of choices."""
+        max_elems = 0
+        largest_list = ""
+        for k, v in self.dict().items():
+            options = v.get("options", [])
+            if k == "disk_request":
+                continue
+            if len(options) > max_elems:
+                max_elems = len(options)
+                largest_list = k
+        return largest_list
+
+
+class _ClassNameIter:
+    def __init__(self):
+        self._vals = ["small", "medium", "large"]
+        self._current_ind = 0
+
+    def __iter__(self):
+        self._current_ind = 0
+        return self
+
+    def __next__(self):
+        curr_ind = self._current_ind
+        self._current_ind += 1
+        if curr_ind >= len(self._vals):
+            return ("x" * (curr_ind - len(self._vals) + 1)) + self._vals[-1]
+        return self._vals[curr_ind]
+
+
+def generate_default_resource_pool(
+    server_options: ServerOptions, defaults: ServerOptionsDefaults
+) -> models.ResourcePool:
+    """Generate a resource pool from the notebook service style server options."""
+    clses: Set[models.ResourceClass] = set()
+    largest_attribute = server_options.find_largest_attribute()
+    options_xref = {
+        "cpu_request": "cpu",
+        "mem_request": "memory",
+        "gpu_request": "gpu",
+    }
+    class_names = _ClassNameIter()
+    largest_attribute_options = getattr(getattr(server_options, largest_attribute), "options")
+    max_storage = round(sorted(server_options.disk_request.options)[-1] / 1_000_000_000)
+    for ival, val in sorted(enumerate(largest_attribute_options)):
```

Comment:
I think it should be `enumerate(sorted(...))` otherwise it just sorts by the enumerate, i.e. it stays in the same order

Reply:
Yes you are right. I fixed it.